soccer/Processor.cpp

```diff
@@ -386,6 +386,16 @@ void Processor::run() {
             delete packet;
         }
 
+        // Log referee data
+        vector<NewRefereePacket*> refereePackets;
+        _refereeModule.get()->getPackets(refereePackets);
+        fprintf(stderr, "Ref packets received this loop:%lu\n",
+                refereePackets.size());
+        for (NewRefereePacket* packet : refereePackets) {
+            SSL_Referee* log = _state.logFrame->add_raw_refbox();
+            log->CopyFrom(packet->wrapper);
+        }
```

Comment:
The best thing would probably be to use `std::shared_ptr`.  It basically adds reference-counting so that when all of the `shared_ptr`s themselves go out of scope, the object that they point to gets `delete`d.

If you're not familiar with C++ smart pointers, I can help you out here.  The main thing would be to change the declaration of the packet vector in NewRefereeModule.hpp from `std::vector<NewRefereePacket*>` to `std::vector<std::shared_ptr<NewRefereePacket>>` and to change the allocation of the packets from `NewRefereePacket* packet = new NewRefereePacket` to `auto packet = make_shared<NewRefereePacket>()`.

Also, I agree... `NewRefereePacket` should really be `RefereePacket` - I think that naming is left over from when we converted from new to old.